dist/jquery.flot.js

```diff
@@ -2459,13 +2459,33 @@ Licensed under the MIT license.
             return precision;
         }
 
-        function setEndpointTicks(axis) {
-            if (axis.options.showTickLabels === 'all' || axis.options.showTickLabels === 'endpoints') {
+        function setEndpointTicks(axis, series) {
+            if (isValidEndpointTick(axis, series)) {
                 axis.ticks.unshift(newTick(axis.min, null, axis, 'min'));
                 axis.ticks.push(newTick(axis.max, null, axis, 'max'));
             }
         }
 
+        function isValidEndpointTick(axis, series) {
+            if (axis.options.showTickLabels === 'endpoints') {
+                return true;
+            }
+            if (axis.options.showTickLabels === 'all') {
+                var axisSeries = series.filter(function(series) {
+                    return series.xaxis === axis || series.yaxis === axis;
+                });
+
+                //if only bars, do not put endpoint ticks
+                for (var i = 0; i < axisSeries.length; i++) {
+                    if (!axisSeries[i].bars.show) {
```

Comment:
I think it is possible to have a series with both lines and bars, or lines and points, etc. How should the ticks be generated in this case?

Reply:
This thing is actually covered in my second test. I consider that if there are multiple types of series, we should stick to the users choice, so we should add the endpoints ticks